scheds/rust/scx_chaos/src/main.rs

```diff
@@ -241,6 +237,30 @@ fn main() -> Result<()> {
         }
     });
 
+
+    if let Some(pid) = args.pid {
+        info!("Monitoring process with PID: {}", pid);
+
+        let is_process_running = |pid: libc::pid_t| -> bool {
+            unsafe {
+                // SAFETY: kill with signal 0 only runs validity checks. There's noc hance of
```

Comment:
s/noc hance/no chance

Reply:
Why kill vs using something like `waitpid`?

Reply:
> Why kill vs using something like `waitpid`?

By memory you can only `waitpid` a child and afaik Linux doesn't have a call to do anything smarter for non-child processes.